### PR TITLE
[front] - feature: update workspace providers

### LINKED ISSUE #5558: [Model Deactivation] - Endpoint for whiteListedProviders & defaultEmbedder update

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at modifying the `ProviderManagement` and workspace handler to update `whiteListedProviders` in DB.

## Risk

None as the modal is not being used yet.

## Deploy Plan

Not to be deployed yet

### CODE REVIEW

File: front/pages/api/w/[wId]/index.ts [hunk 1]
```diff
@@ -25,10 +26,20 @@ const WorkspaceAllowedDomainUpdateBodySchema = t.type({
   domainAutoJoinEnabled: t.boolean,
 });
 
+const WorkspaceProvidersUpdateBodySchema = t.type({
+  whiteListedProviders: t.array(t.string),
```
Comment (PopDaph) [inline front/pages/api/w/[wId]/index.ts:30]: Not using `ModelProviderIdType` ?

File: front/pages/api/w/[wId]/index.ts [hunk 2]
```diff
@@ -25,10 +26,20 @@ const WorkspaceAllowedDomainUpdateBodySchema = t.type({
...
+});
+
+const WorkspaceEmbedderUpdateBodySchema = t.type({
+  defaultEmbeddingProvider: t.string,
```
Comment (PopDaph) [inline front/pages/api/w/[wId]/index.ts:34]: Same here, why not using `ModelProviderIdType` ?

File: front/pages/api/w/[wId]/index.ts [hunk 3]
```diff
@@ -103,6 +114,20 @@ async function handler(
         });
 
         owner.ssoEnforced = body.ssoEnforced;
+      } else if ("whiteListedProviders" in body) {
+        if (body.whiteListedProviders.every(isModelProviderId)) {
```
Comment (PopDaph) [inline front/pages/api/w/[wId]/index.ts:118]: We should be able to remove this check and the one on isModelProviderId if we use the valid types on `WorkspaceProvidersUpdateBodySchema` & `WorkspaceEmbedderUpdateBodySchema`?

File: types/src/front/lib/assistant.ts [hunk 4]
```diff
@@ -18,7 +18,8 @@ export const isModelProviderId = (
 ): providerId is ModelProviderIdType =>
   MODEL_PROVIDER_IDS.includes(providerId as ModelProviderIdType);
 
-export const ModelProviderIdCodec = ioTsEnum(MODEL_PROVIDER_IDS);
+export const ModelProviderIdCodec =
```
Comment (PopDaph) [inline types/src/front/lib/assistant.ts:21]: NIT: can we do the same for ModelIdCodec below?

**Review by PopDaph — APPROVED** [review]
🙏🏻

File: types/src/front/lib/assistant.ts (hunk 4 above)
Comment (JulesBelveze) [reply types/src/front/lib/assistant.ts:21]: ✅ 😃

**Pull request merged**